app/controllers/api/v1/users/events_controller.rb

```diff
@@ -3,13 +3,14 @@
 class Api::V1::Users::EventsController < ApplicationController
   def index
     user = User.find(params[:user_id])
-    # TODO: add flexibility based on query params to retrieve upcoming vs past
-    # events (e.g. if we decide to add activity log to dashboard)
-    events = user.upcoming_events
+    events = if params[:time_frame] == 'past'
+               user.past_events(params[:user_id])
+             else
+               user.upcoming_events(params[:user_id])
+             end
 
-    options = {}
-    options[:include] = [:user]
-    render json: EventSerializer.new(events, options).serializable_hash,
-           status: :ok
+    render json: EventSerializer.new(
+      events, { params: { current_user: params[:user_id] } }
+    ).serializable_hash, status: :ok
```

Comment:
This was such a fun solution!  I think we can use this same process for some other endpoints as well.  Will make things more flexible for consuming on the FE.

Reply:
🤘🏽thank you for the help working through it!!